docs/proposals/network-policy.md

```diff
@@ -0,0 +1,174 @@
+- A v1beta NetworkPolicy API object
+- A field on the `Namespace` object to control isolation.
+
+### Namespace changes
+
+> Note that this section is subject to some change with the addition of egress policy as mentioned above.
+
+The `Namespace` object will be augmented with the field `networkIsolation` which will take one of the following values:
+
+When `networkIsolation=yes` is set on a namespace:
+- Pods in that Namespace will not be accessible from any other source, unless explicitly allowed by a NetworkPolicy object.
+- Pods in that Namespace will be able to access any other source (egress is uninhibited).
+
+When `networkIsolation=no` is set on a namespace:
+- Pods in that Namespace will be accessible from any other source.
+- Pods in that Namespace will be able to access any other source.
+
+The `networkIsolation` field will default to `false` if not specified.
+
+### NetworkPolicy YAML Definition and Behavior
+
+Once the namespace is isolated, a mechanism to selectively allow traffic into the namespace and between pods within
+the namespace is required.  That is accomplished through ingress rules on `NetworkPolicy`
+objects (of which there can be multiple in a single namespace).  Pods selected by
+a NetworkPolicy should allow any incoming connections that match any
+ingress rule, per the network plugin’s capabilities.
+
+```yaml
+kind: NetworkPolicy
+apiVersion: v1beta 
+metadata:
+  name:
+  namespace:
+spec:
+  podSelector:            // Standard label selector - selects pods.  
+  ingress:                // List of ingress rules (optional).             
+    - ports:              // List of allowed ports / protocols (optional).          
+        - port:           // Port on the specified protocol (optional). 
+          protocol:       // Protocol (TCP, UDP) 
+      from:               // List of allowed sources (optional).    
+        - pods:           // Label selector - selects Pods (optional). 
+          namespaces:     // Label selector - selects Namespaces (optional).
+```
+
+Each NetworkPolicy object supports the following fields:
+- `podSelector`: A label selector which selects which pods this `NetworkPolicy` applies to.
+- `ingress`: A list of ingress rules.  If not present, this `NetworkPolicy` has no effect on ingress to the selected pods.
+
+Each ingress rule supports the following fields:
+- `ports`: A list of protocol / ports which should be accessible on the selected pods. If not defined, then access to the selected pods will not be restricted by port/protocol.
+- `from`: A list of source criteria which dictate which sources should be able to access the pods selected. If no `from` key is provided, then access will not be restricted based on source.
+
+Each item in the `ports` list supports the following fields:
+- `protocol`: The protocol for the given port - TCP or UDP.
+- `port`: The port on the specified protocol.
+
+Each item in the `from` list may declare at most one of the following:
```

Comment:
Should clarify that matching must match pods AND namespaces if both are present before allowing traffic.  OR can be handled with multiple ingress elements.